crates/ruff_python_semantic/src/model.rs

```diff
@@ -257,6 +257,29 @@ impl<'a> SemanticModel<'a> {
             .is_some_and(|binding| binding.kind.is_builtin())
     }
 
+    /// Return `true` if `member` is a reference to `builtins.$target`,
+    /// i.e. either `object` (where `object` is not overridden in the global scope),
+    /// or `builtins.object` (where `builtins` is imported as a module at the top level)
+    #[allow(clippy::inline_always)]
+    #[inline(always)]
```

Comment:
Let me know if this isn't the right thing to do here, but it improved performance a ton locally...

Reply:
But apparently not enough to make much of a dent on how codspeed sees things... https://codspeed.io/astral-sh/ruff/branches/AlexWaygood:improve-builtins-handling

Reply:
Ah yeah, I don't think this should be necessary. If anything you might consider finding a way to mark the `if seen(BUILTINS)` branch as cold, but we don't really do that anywhere else.